packages/eslint-plugin-wpcalypso/lib/rules/jsx-classname-namespace.js

```diff
@@ -43,74 +41,41 @@ const rule = ( module.exports = function ( context ) {
 		return false;
 	}
 
-	function isFolderRootFile( filename ) {
-		return rootFiles.indexOf( path.basename( filename ) ) !== -1;
-	}
-
 	return {
 		JSXAttribute: function ( node ) {
 			if ( 'className' !== node.name.name ) {
 				return;
 			}
-
 			let rawClassName;
 			if ( 'JSXExpressionContainer' === node.value.type ) {
 				rawClassName = node.value.expression;
 			} else {
 				rawClassName = node.value;
 			}
-
 			if ( 'Literal' !== rawClassName.type || 'string' !== typeof rawClassName.value ) {
 				return;
 			}
-
 			// we don't validate elements inside `ReactDOM.render` expressions
 			if ( isInRenderCallExpession( node ) ) {
 				return;
 			}
 
-			const filename = context.getFilename();
-
-			const namespaces = [ path.basename( path.dirname( filename ) ) ];
-			if ( ! isFolderRootFile( filename ) ) {
-				namespaces.push( path.basename( filename, path.extname( filename ) ) );
-			}
-
-			const prefixPatterns = namespaces.map(
-				( namespace ) => new RegExp( `^${ namespace }(__[a-z0-9-]+)?$` )
-			);
-
+			// Extract class names into an array.
 			const classNames = rawClassName.value.split( ' ' );
-			const isError = ! classNames.some( ( className ) =>
-				prefixPatterns.some( ( prefixPattern ) => prefixPattern.test( className ) )
-			);
-
+			const isError = ! classNames.some( ( className ) => namePattern.test( className ) );
 			if ( ! isError ) {
 				return;
 			}
 
-			const expected =
-				namespaces.map( ( namespace ) => namespace + '__' ).join( ' or ' ) + ' prefix';
-
 			context.report( {
```

Comment:
Is there any way to keep "expected" classnames?

Reply:
I can put it as part of the error message, but it can become rather long. 
Example:
```
className should adhere to BEM convention, like foo__bar, foo-bar, foo__bar-baz.
```

Reply:
Fair enough 👍 

I guess it would be impossible to magically turn the class name into the right one.